lib/core/src/Cardano/Wallet/TokenMetadata.hs

```
@@ -401,14 +402,29 @@ nullTokenMetadataClient :: Applicative m => TokenMetadataClient m
 nullTokenMetadataClient = TokenMetadataClient $ \_ ->
     pure . Right $ BatchResponse []
 
+-- | Mock metadata client using a hard-coded server response from a file for
+-- data.
+mockFileTokenMetadataClient :: URI -> TokenMetadataClient IO
+mockFileTokenMetadataClient uri = TokenMetadataClient $ \(BatchRequest subjects _)-> do
+    let filePath = T.unpack
+            . fromMaybe (error "mockFileTokenMetadataClient: bad file path")
```

😬 Maybe we should just fail silently, and at most log a error.